src/AutoMapper/XpressionMapper/Extensions/MapperExtensions.cs

```diff
@@ -33,9 +34,15 @@ public static TDestDelegate MapExpression<TDestDelegate>(this IMapper mapper, La
             var remappedBody = visitor.Visit(expression.Body);
             if (remappedBody == null)
                 throw new InvalidOperationException(Resource.cantRemapExpression);
+
+            return (TDestDelegate)Lambda
+                                    (
+                                        typeDestFunc,
+                                        ExpressionFactory.ToType(remappedBody, typeDestFunc.GetGenericArguments().Last()),
+                                        expression.GetDestinationParameterExpressions(visitor.InfoDictionary, typeMappings)
+                                    );
+        }
 

```

Comment:
The two MapExpression methods are almost the same, only the visitor differs. Some code reuse seems in order.

Reply:
I agree. The simplest method would be to pass a `Func<Dictionary<Type, Type>, XpressionMapperVisitor> func` to the consolidated method.  That belongs in a separate PR though - we did not add the two methods in this PR -  unless you want to fix it.

Reply:
Nah, I'm good :)